File: seed/csharp-model/response-property/src/SeedResponseProperty/Service/Response.cs
using System.Text.Json.Serialization;
using SeedResponseProperty;

#nullable enable

namespace SeedResponseProperty;

public class Response
{
    [JsonPropertyName("data")]
    public Movie Data { get; init; }

    [JsonPropertyName("metadata")]
    public Dictionary<string, string> Metadata { get; init; }

    [JsonPropertyName("docs")]
    public string Docs { get; init; }
}

File: seed/php-model/response-property/src/Service/Response.php
<?php

namespace Seed\Service;

use Seed\Core\Json\SerializableType;
use Seed\Traits\WithMetadata;
use Seed\Service\Traits\WithDocs;
use Seed\Core\Json\JsonProperty;

class Response extends SerializableType
{
    use WithMetadata;
    use WithDocs;

    /**
     * @var Movie $data
     */
    #[JsonProperty('data')]
    public Movie $data;

    /**
     * @param array{
     *   data: Movie,
     *   metadata: array<string, string>,
     *   docs: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->data = $values['data'];
        $this->metadata = $values['metadata'];
        $this->docs = $values['docs'];
    }
}
